From 2997bec965fd1d5c0ba41c984d2351709a418a73 Mon Sep 17 00:00:00 2001
Message-Id: <2997bec965fd1d5c0ba41c984d2351709a418a73.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Wed, 28 Jan 2015 09:12:32 -0500
Subject: [PATCH 84/95] MUST_REBASE [VPG]: drm/i915/adf: Re-Organize the get &
 put calls for chv

S0ix VLV WA is not true for CHV. Re-organizing the get/put calls to get
support chv s0ix

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need
to work on i915 and ADF convergence path before it can be upstreamed

Issue: GMINL-5501
Change-Id: I86eca97eec0e636f671d4a352ec4d5dc293cfebf
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_adf_wrapper.c |   22 ++++++++++++++++++----
 1 file changed, 18 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_adf_wrapper.c b/drivers/gpu/drm/i915/i915_adf_wrapper.c
index 80351d2..1a6597f 100644
--- a/drivers/gpu/drm/i915/i915_adf_wrapper.c
+++ b/drivers/gpu/drm/i915/i915_adf_wrapper.c
@@ -53,7 +53,13 @@ static struct drm_i915_private *i915_adf_dev;
 void i915_adf_wrapper_init(struct drm_i915_private *dev_priv)
 {
 	i915_adf_dev = dev_priv;
-	intel_runtime_pm_get(dev_priv);
+
+	/*
+	 * rpm_get for vlv due is to HW WA, we do not do rpm_get in driver load
+	 * In chv, WA is not applicable so we ignore this as rpm_get.
+	 */
+	if (IS_VALLEYVIEW(dev_priv->dev) && !IS_CHERRYVIEW(dev_priv->dev))
+		intel_runtime_pm_get(dev_priv);
 }
 
 void i915_adf_wrapper_teardown(void)
@@ -104,8 +110,11 @@ void intel_adf_display_rpm_get(void)
 		return;
 
 	dev_priv = i915_adf_dev;
-	intel_runtime_pm_get(dev_priv);
-	intel_display_set_init_power(dev_priv, true);
+
+	if (IS_VALLEYVIEW(dev_priv->dev) && !IS_CHERRYVIEW(dev_priv->dev))
+		intel_runtime_pm_get(dev_priv);
+
+	intel_display_power_get(dev_priv, POWER_DOMAIN_INIT);
 }
 EXPORT_SYMBOL(intel_adf_display_rpm_get);
 
@@ -117,7 +126,12 @@ void intel_adf_display_rpm_put(void)
 		return;
 
 	dev_priv = i915_adf_dev;
-	intel_runtime_pm_put(dev_priv);
+
+	if (IS_VALLEYVIEW(dev_priv->dev) && !IS_CHERRYVIEW(dev_priv->dev))
+		intel_runtime_pm_put(dev_priv);
+
+	intel_display_set_init_power(dev_priv, false);
+	intel_display_power_put(dev_priv, POWER_DOMAIN_INIT);
 }
 EXPORT_SYMBOL(intel_adf_display_rpm_put);
 
-- 
1.7.9.5

